docs/service-discovery/dns.md

```diff
@@ -0,0 +1,46 @@
+# DNS Service Discovery
+
+DNS is a name resolution protocol used to determine IP addresses for hostnames.
+That makes it a natural fit for service discovery. Among others, Consul and AWS Cloud Map provide DNS APIs for service discovery.
+
+This page explains how Stork can use DNS to handle the service discovery.
+
+By default, DNS service discovery uses _SRV_ records to discover service locations. The target of the _SRV_ record is
+resolved to IP by querying for _A_ and _AAAA_ records.
+
+## Dependency
+
+To use the DNS service discovery, you need to add the Stork DNS Service Discovery provider dependency to your project:
+
+```xml
+<dependency>
+    <groupId>io.smallrye.stork</groupId>
+    <artifactId>stork-service-discovery-dns</artifactId>
+    <version>{{version.current}}</version>
+</dependency>
+```
+
+## Configuration
+
+Next, set the service discovery `type` to `DNS`.
+Additionally, you would usually specify the DNS server, or servers, to use for the discovery. 
+All in all, your configuration could look as follows:
+
+=== "stork standalone"
+    ```properties
+    stork.my-service.service-discovery.type=dns
+    # optional dns servers:
+    stork.my-service.service-discovery.dns-servers=my-dns-server:8221,my-dns-server2
+    ```
+
+=== "stork in quarkus"
+    ```properties
+    quarkus.stork.my-service.service-discovery.type=dns
+
+    # optional dns servers:
+    quarkus.stork.my-service.service-discovery.dns-servers=my-dns-server:8221,my-dns-server2
+    ```
+
```

Comment:
How do you name the service? I believe in the previous example it's `my=service`, can it be overridden? 

Also, I believe a section explaining the resolution in detail would be nice:

1. resolve the `my-service` SRV record (and show the content of the record)
2. resolve the IP from the indicated names

What about multiple entries in the SRV and A / AAA records.

Reply:
Yes, but it. needs to be written in the doc. For someone that does not understand how DNS works, it can be very confusing to have two resolutions.

Reply:
I added a few sentences and a pic:
![image](https://user-images.githubusercontent.com/3901322/179616473-1a673995-d7ef-4d55-8efc-7ae10a1ca4b7.png)

Reply:
Thanks!